contracts/core/RebaseController.sol

```diff
@@ -6,163 +6,143 @@ import "@openzeppelin/contracts/access/Ownable.sol";
-                    strategiesAdded++;
-                }
-            }
+    /**
+     * @notice Pauses the priority pool if a loss has been detected
+     * @param _performData abi encoded index of strategy with a loss
+     */
+    function performUpkeep(bytes calldata _performData) external {
+        if (priorityPool.poolStatus() == IPriorityPool.PoolStatus.CLOSED) revert PoolClosed();
 
-            return (true, abi.encode(strategyIdxs, uint256(-1 * totalDepositChange)));
-        }
+        uint256 strategyIdxWithLoss = abi.decode(_performData, (uint256));
+        address[] memory strategies = stakingPool.getStrategies();
 
-        return (false, "0x");
+        if (IStrategy(strategies[strategyIdxWithLoss]).getDepositChange() >= 0)
+            revert NoLossDetected();
+
+        priorityPool.setPoolStatus(IPriorityPool.PoolStatus.CLOSED);
+        if (address(securityPool) != address(0)) securityPool.initiateClaim();
     }
 
     /**
-     * @notice Updates rewards in the case of a negative rebase or pauses the priority
-     * pool if losses exceed the maximum
-     * @dev should be called by a custom bot (not CL automation)
-     * @param _performData abi encoded list of strategy indexes to update and their total deposit change
+     * @notice Pauses the priority pool in the case of an emergency
      */
-    function performUpkeep(bytes calldata _performData) external onlyRebaseBot {
+    function pausePool() external onlyEmergencyPauser {
         if (priorityPool.poolStatus() == IPriorityPool.PoolStatus.CLOSED) revert PoolClosed();
 
-        (uint256[] memory strategiesToUpdate, uint256 totalDepositChange) = abi.decode(
-            _performData,
-            (uint256[], uint256)
-        );
-
-        if (strategiesToUpdate.length == 0 || totalDepositChange == 0)
-            revert NoStrategiesToUpdate();
-
-        if ((10000 * totalDepositChange) / stakingPool.totalSupply() > maxRebaseLossBP) {
-            priorityPool.setPoolStatus(IPriorityPool.PoolStatus.CLOSED);
-            securityPool.initiateClaim();
-        } else {
-            stakingPool.updateStrategyRewards(strategiesToUpdate, "");
-        }
+        priorityPool.setPoolStatus(IPriorityPool.PoolStatus.CLOSED);
+        if (address(securityPool) != address(0)) securityPool.initiateClaim();
     }
 
     /**
      * @notice Reopens the priority pool and security pool after they were paused as a result
-     * of a significant slashing event and rebases the staking pool
-     * @dev sender should ensure all strategies with losses are included in the index list and
-     * all strategies with gains are excluded
-     * @param _strategyIdxs list of strategy indexes to update
+     * of a loss and updates strategy rewards in the staking pool
+     * @param _data encoded data to pass to strategies
      */
-    function reopenPool(uint256[] calldata _strategyIdxs) external onlyOwner {
+    function reopenPool(bytes calldata _data) external onlyOwner {
```

Comment:
Add a check so this can only be called if pool is closed?

Reply:
Priority pool will enforce that but better to be explicit so I'll add a check